packages/paste-core/components/ai-chat-log/src/AIChatMessageLoading.tsx

```diff
@@ -0,0 +1,97 @@
+import { Box, safelySpreadBoxProps } from "@twilio-paste/box";
+import type { BoxElementProps } from "@twilio-paste/box";
+import { Button } from "@twilio-paste/button";
+import { StopIcon } from "@twilio-paste/icons/esm/StopIcon";
+import { ScreenReaderOnly } from "@twilio-paste/screen-reader-only";
+import { SkeletonLoader } from "@twilio-paste/skeleton-loader";
+import type { HTMLPasteProps } from "@twilio-paste/types";
+import * as React from "react";
+
+import { AIMessageContext } from "./AIMessageContext";
+
+const clampedRandom = (min: number, max: number): number => {
+  return Math.min(Math.max(min, Math.random() * max), max);
+};
+export interface AIChatMessageLoadingProps extends HTMLPasteProps<"div"> {
+  children?: never;
+  /**
+   * Overrides the default element name to apply unique styles with the Customization Provider
+   *
+   * @default "AI_CHAT_MESSAGE_LOADING"
+   * @type {BoxProps["element"]}
+   * @memberof AIChatMessageLoadingProps
+   */
+  element?: BoxElementProps["element"];
+  /**
+   * Function to call when "Stop generating AI response" button is clicked. If not provided, the button will not be rendered.
+   *
+   * @type {() => void}
+   * @memberof AIChatMessageLoadingProps
+   */
+  onStopLoading?: () => void;
+  /**
+   * I18n label text for the "Stop generating" button when `onStopLoading` is passed for non-english languages.
+   *
+   * @default "Stop generating"
+   * @type {string}
+   * @memberof AIChatMessageLoadingProps
+   */
+  i18nStopGeneratingLabel?: string;
+  /**
+   * I18n accessible screen reader text to give context to the "Stop generating" button when `onStopLoading` is passed for non-english languages. Should read as a sentence, e.g. "Stop generating AI response".
+   *
+   * @default "AI response"
+   * @type {string}
+   * @memberof AIChatMessageLoadingProps
+   */
+  i18nAIScreenReaderText?: string;
+}
+
+export const AIChatMessageLoading = React.forwardRef<HTMLDivElement, AIChatMessageLoadingProps>(
+  (
+    {
+      onStopLoading,
+      i18nStopGeneratingLabel = "Stop generating",
+      i18nAIScreenReaderText = "AI response",
```

Comment:
Should the default text here match the props description? "Stop generating AI response"

Reply:
It's kind of clunky but I made them 2 separate props - stop generating label and screen reader text. Only the stop generating label gets rendered on the screen, but the text passed into the screen reader text prop goes into a Screen Reader Only component to give screen reader users more context about what stops generating. So a sighted user will see "Stop generating" but a screen reader will read out loud "stop generating AI response"